ehrapy/api/encode/encode.py

```diff
@@ -0,0 +1,464 @@
+            if len(categoricals) != len(set(categoricals)):
+                raise ValueError(
+                    "The categorical column names given contain at least one duplicate column. "
+                    "Check the column names to ensure that no column is encoded twice!"
+                )
+            Encoder._add_categories_to_obs(ann_data, categoricals)
+            Encoder._add_categories_to_uns(ann_data, categoricals)
+            current_encodings = (
+                {} if "current_encodings" not in ann_data.uns.keys() else ann_data.uns["current_encodings"]
+            )
+            encoded_x = None
+            encoded_var_names = ann_data.var_names.to_list()
+
+            for encoding_mode in categoricals_encode_mode.keys():
+                if encoding_mode not in Encoder.available_encodings:
+                    raise ValueError(
+                        f"Unknown encoding mode {encoding_mode}. Please provide one of the following encoding modes:\n"
+                        f"{Encoder.available_encodings}"
+                    )
+                encode_mode_switcher = {
+                    "one_hot_encoding": Encoder._one_hot_encoding,
+                    "label_encoding": Encoder._label_encoding,
+                    "count_encoding": Encoder._count_encoding,
+                }
+                # perform the actual encoding
+                encoded_x, encoded_var_names = encode_mode_switcher[encoding_mode](
+                    ann_data, encoded_x, encoded_var_names, categoricals_encode_mode[encoding_mode]
+                )
+                # update encoding history in uns
+                for categorical in categoricals_encode_mode[encoding_mode]:
+                    current_encodings[categorical] = encoding_mode
+
+            # update original layer content with the new categorical encoding and the old other values
+            updated_layer = Encoder._update_layer_after_encode(
+                ann_data.layers["original"],
+                encoded_x,
+                encoded_var_names,
+                ann_data.var_names.to_list(),
+                categoricals,
+            )
+            try:
+                encoded_ann_data = AnnData(
+                    encoded_x,
+                    obs=ann_data.obs.copy(),
+                    var=dict(var_names=encoded_var_names),
+                    uns=ann_data.uns.copy(),
+                    layers={"original": updated_layer},
+                )
+                # update current encodings in uns
+                encoded_ann_data.uns["current_encodings"] = current_encodings
+
+            # if the user did not pass every non numerical column for encoding, a Anndata object cannot be created
+            except ValueError:
+                print(
+                    "[bold red]Creation of AnnData object failed. "
+                    "Ensure that you passed all non numerical, categorical values for encoding!"
+                )
+                sys.exit(1)
```

Comment:
This might be bad when ehrapy is used as a library in notebooks as it is designed to, right?
Let's not use sys.exits because this is not just a CLI app